### PR TITLE
Dockerfile - Add ROCm6.2 dockerfile

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
Add ROCm6.2 dockerfile.

### CODE REVIEW

File: dockerfile/rocm6.2.x.dockerfile [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+RUN cp -r /opt/superbench/third_party/hipBLASLt/build/release/hipblaslt-install/lib/*  /opt/rocm/lib/ && \
+    cp -r /opt/superbench/third_party/hipBLASLt/build/release/hipblaslt-install/include/*  /opt/rocm/include/
+RUN cd third_party/Megatron/Megatron-DeepSpeed && \
+    git apply ../megatron_deepspeed_rocm6.patch
+
+# Install transformer_engine
+git clone --recursive https://github.com/ROCm/TransformerEngine.git
+cd TransformerEngine
+export NVTE_FRAMEWORK=pytorch #optionally set framework, currently only support pytorch and jax
+export PYTORCH_ROCM_ARCH=gfx942 # CK fused attn only support MI200 and MI300 and fp8 features are only supported on MI300
```
Comment (yzygitzh) [inline dockerfile/rocm6.2.x.dockerfile:187]: Does this mean that this docker image only supports mi300x?

File: .github/workflows/build-image.yml [hunk 2]
```diff
@@ -44,12 +44,12 @@ jobs:
           dockerfile: rocm5.7.x
           tags: superbench/main:rocm5.7
           runner: [self-hosted, rocm-build]
-          build_args: "NUM_MAKE_JOBS=64"
-        - name: rocm6.0
-          dockerfile: rocm6.0.x
-          tags: superbench/main:rocm6.0
+          build_args: "NUM_MAKE_JOBS=16"
+        - name: rocm6.2
+          dockerfile: rocm6.2.x
```
Comment (guoshzhao) [inline .github/workflows/build-image.yml:49]: We only release the latest rocm image?

Comment (yukirora) [reply .github/workflows/build-image.yml:48]: we will release one old(rocm6.0) and one latest(rocm6.2)

File: dockerfile/rocm6.2.x.dockerfile (hunk 1 above)
Comment (yukirora) [reply dockerfile/rocm6.2.x.dockerfile:187]: yes, for other mi100 or older gpus, they can use rocm6.0 instead

**Review by guoshzhao — APPROVED** [review]

**Review by yzygitzh — APPROVED** [review]

**Pull request merged**